### PR TITLE
[Merged by Bors] - feat(Order/Filter/Basic): eventually_iff_all_subsets and specializations

### PR DESCRIPTION
**Pull request by gaetanserre**
The set `{x | p x}` belongs to `f` iff `∀ s, {x | x ∈ s → p x} ∈ f`. Specialization for `EventuallyEq` and `Eventually.LE`.

Suggested by @urkud.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Filter/Basic.lean [hunk 1]
```diff
@@ -1688,6 +1697,10 @@ theorem eventuallyLE_congr {f f' g g' : α → β} (hf : f =ᶠ[l] f') (hg : g =
   ⟨fun H => H.congr hf hg, fun H => H.congr hf.symm hg.symm⟩
 #align filter.eventually_le_congr Filter.eventuallyLE_congr
 
+theorem eventuallyLE_iff_all_subsets {f g : α → β} {l : Filter α} :
+    f ≤ᶠ[l] g ↔ ∀ (s : Set α), ∀ᶠ x in l, x ∈ s → f x ≤ g x :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Filter/Basic.lean:1701]:
```suggestion
    f ≤ᶠ[l] g ↔ ∀ s : Set α, ∀ᶠ x in l, x ∈ s → f x ≤ g x :=
```

File: Mathlib/Order/Filter/Basic.lean [hunk 2]
```diff
@@ -1666,6 +1671,10 @@ theorem eventuallyEq_iff_sub [AddGroup β] {f g : α → β} {l : Filter α} :
   ⟨fun h => h.sub_eq, fun h => by simpa using h.add (EventuallyEq.refl l g)⟩
 #align filter.eventually_eq_iff_sub Filter.eventuallyEq_iff_sub
 
+theorem eventuallyEq_iff_all_subsets {f g : α → β} {l : Filter α} :
+    f =ᶠ[l] g ↔ ∀ (s : Set α), ∀ᶠ x in l, x ∈ s → f x = g x :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Filter/Basic.lean:1675]:
```suggestion
    f =ᶠ[l] g ↔ ∀ s : Set α, ∀ᶠ x in l, x ∈ s → f x = g x :=
```

File: Mathlib/Order/Filter/Basic.lean [hunk 3]
```diff
@@ -1279,6 +1279,11 @@ theorem eventually_inf_principal {f : Filter α} {p : α → Prop} {s : Set α}
   mem_inf_principal
 #align filter.eventually_inf_principal Filter.eventually_inf_principal
 
+theorem eventually_iff_all_subsets {f : Filter α} {p : α → Prop} :
+    (∀ᶠ x in f, p x) ↔ ∀ (s : Set α), ∀ᶠ x in f, x ∈ s → p x :=
+  ⟨(fun h _ ↦ by filter_upwards [h] with _ pa _ using pa),
+  (fun h ↦ by filter_upwards [h univ] with _ pa using (pa (by simp)))⟩
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Filter/Basic.lean:1285]: This is quite hard to read with all the brackets, so I'd recommend:
```suggestion
    (∀ᶠ x in f, p x) ↔ ∀ s : Set α, ∀ᶠ x in f, x ∈ s → p x where
  mp h _ := by filter_upwards [h] with _ pa _ using pa
  mpr h := by filter_upwards [h univ] with _ pa using (pa (by simp))
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Sorry for the delay here. This looks sensible to me; just some formatting comments.

File: Mathlib/Order/Filter/Basic.lean (hunk 3 above)
Comment (gaetanserre) [reply Mathlib/Order/Filter/Basic.lean:1285]: Oh nice, I did not know we could do that!

Comment (gaetanserre) [reply Mathlib/Order/Filter/Basic.lean:1285]: Why does it works? Doesn't this notation refer to typeclasses?

File: Mathlib/Order/Filter/Basic.lean [hunk 4]
```diff
@@ -1279,6 +1279,11 @@ theorem eventually_inf_principal {f : Filter α} {p : α → Prop} {s : Set α}
   mem_inf_principal
 #align filter.eventually_inf_principal Filter.eventually_inf_principal
 
+theorem eventually_iff_all_subsets {f : Filter α} {p : α → Prop} :
+    (∀ᶠ x in f, p x) ↔ ∀ (s : Set α), ∀ᶠ x in f, x ∈ s → p x where
+    mp h _ := by filter_upwards [h] with _ pa _ using pa
+    mpr h := by filter_upwards [h univ] with _ pa using pa (by simp)
```
Comment (jcommelin) [inline Mathlib/Order/Filter/Basic.lean:1285]:
```suggestion
  mp h _ := by filter_upwards [h] with _ pa _ using pa
  mpr h := by filter_upwards [h univ] with _ pa using pa (by simp)
```

**Comment by jcommelin** [comment]
My apologies that this PR went under the radar for so long. An automated check was failing for no good reason. @grunweg created some infrastructure to catch such cases in the future: https://jcommelin.github.io/queueboard/help_out.html#inessential-CI-fails

**Comment by gaetanserre** [comment]
> My apologies that this PR went under the radar for so long. An automated check was failing for no good reason. @grunweg created some infrastructure to catch such cases in the future: https://jcommelin.github.io/queueboard/help_out.html#inessential-CI-fails

Thank you very much!

**Pull request closed without merging**